libp2p/dialer.nim

```diff
@@ -54,53 +56,61 @@ proc dialAndUpgrade(
     transport: Transport
     address: MultiAddress
 
-  for t in self.transports: # for each transport
-    transport = t
-    for a in addrs:      # for each address
-      address = a
-      if t.handles(a):   # check if it can dial it
-        trace "Dialing address", address = $a, peerId
-        let dialed = try:
-            libp2p_total_dial_attempts.inc()
-            # await a connection slot when the total
-            # connection count is equal to `maxConns`
-            await self.connManager.trackOutgoingConn(
-              () => transport.dial(address)
-            )
-          except TooManyConnectionsError as exc:
-            trace "Connection limit reached!"
-            raise exc
-          except CancelledError as exc:
-            debug "Dialing canceled", msg = exc.msg, peerId
-            raise exc
-          except CatchableError as exc:
-            debug "Dialing failed", msg = exc.msg, peerId
-            libp2p_failed_dials.inc()
-            continue # Try the next address
-
-        # make sure to assign the peer to the connection
-        dialed.peerId = peerId
-
-        # also keep track of the connection's bottom unsafe transport direction
-        # required by gossipsub scoring
-        dialed.transportDir = Direction.Out
-
-        libp2p_successful_dials.inc()
-
-        let conn = try:
-            await transport.upgradeOutgoing(dialed)
-          except CatchableError as exc:
-            # If we failed to establish the connection through one transport,
-            # we won't succeeded through another - no use in trying again
-            await dialed.close()
-            debug "Upgrade failed", msg = exc.msg, peerId
-            if exc isnot CancelledError:
-              libp2p_failed_upgrades_outgoing.inc()
-            raise exc
-
-        doAssert not isNil(conn), "connection died after upgradeOutgoing"
-        debug "Dial successful", conn, peerId = conn.peerId
-        return conn
+  for rawAddress in addrs:      # for each address
+    address = rawAddress
```

Comment:
It doesn't look like we need `var address` from above anymore and we should be able to simplify the for to `for address in addrs: ...`?

Reply:
Good point, this code wasn't working in 1.4 actually, I fixed it